### ISSUE TITLE
Go to component definition is not working with nested components

### ISSUE THREAD

**Issue by robaguilera** [NullVoxPopuli/coc-ember#6]
Environment:
Neovim: v0.4.2
Ember: 3.13 Octane Edition
MacOS v 10.14.6

Steps to Reproduce:
1. Create a nested glimmer component `ember g component foo -gc -ns`
2. Add the nested component to a template `<Foo />`
3. Invoking goto command `(coc-definition)` on nested template gives the error `[coc.nvim] Definition not found`.

Notes:
Everything seems to be working fine as the goto command works as expected with classic and pod components.

**Comment by robaguilera** [comment]
Just discovered something.

When you have a nested component you can invoke it as either `<Foo />` or `<Foo::Index />`.  The goto command works with `<Foo::Index/>` but not `<Foo />`.  Maybe that's good enough, but it would be nice if it could work with either invocation.

**Comment by NullVoxPopuli** [comment]
Thanks for the report!

/Cc @lifeart

**Comment by lifeart** [comment]
@robaguilera thank you for report, could you write you components structure? 

like

‘’’
app/components
    foo.hbs
    foo/index.hbs
‘’’

**Comment by robaguilera** [comment]
@lifeart sure.

```
app
  /components
    /foo
        index.hbs
        index.js
  /pods
    /home
        template.hbs
```

Here's a simplified version of the repo I'm working with: https://github.com/robaguilera/ember-coc-test if that helps.  In that repo you can see the "bug" in action.  In the `home/template.hbs` file you'll see the component AppHeader invoked two times: `<AppHeader />` and `<AppHeader::Index/>`.

**Comment by lifeart** [comment]
@robaguilera thank you for examples!
How did you get `index` in file names? is it generated by octane blueprint?
I beleve `<AppHeader` and `<AppHeader::Index` - is 2 different components.

If you have any reference to `index` names it will be greate.

**Comment by robaguilera** [comment]
@lifeart not a problem!
The `index` in the file name is automatically generated by running `ember g component app-header -gc -ns` . The flag `gc` is for generating a glimmer component and the flag `ns` is for nested structure.
If you run the command `ember help generate component` inside that sample repo I provided  you'll see all the new flags for generating different kinds of components.

**Comment by robaguilera** [comment]
I forgot to add the glimmer library to the repo so just added that in. You'll see that the AppHeader is loaded twice with both kinds of syntax `<AppHeader />` and `<AppHeader::Index />`.  Maybe it's a bug on Ember's part?  Seems like that could cause all kinds of confusion with naming collisions but thought I'd point it out nonetheless.

**Comment by lifeart** [comment]
@NullVoxPopuli, @robaguilera
`lifeart/ember-language-server`
 `component-context-info-origin` branch updated 

https://github.com/lifeart/ember-language-server/commit/a7f3436df1d28c1fe5ae8edea71b035560f4e13d

**Comment by NullVoxPopuli** [comment]
legit, I'll cut a new release of coc-ember

**Comment by NullVoxPopuli** [comment]
@robaguilera can you try the latest release of coc-ember?

**Comment by robaguilera** [comment]
Working perfectly, wow.  Thanks a lot @lifeart and @NullVoxPopuli!

**Comment by lifeart** [comment]
one more autocomplete fix https://github.com/lifeart/ember-language-server/commit/330a8b0dd7ffc3260783bf4017eb306460994bdd

**Comment by NullVoxPopuli** [comment]
included in 1.0.15 :+1:
